core/tests/test_utils.py

```diff
@@ -1076,6 +1288,45 @@ def swap(self, obj, attr, newvalue):
         finally:
             setattr(obj, attr, original)
 
+    @contextlib.contextmanager
+    def swap_conditionally(
+            self, obj, attr, new_function=lambda *_, **__: None,
+            condition=None):
+        """Swap the obj.attr function to return a value when a condition is met.
+
+        When the condition is not met, then the original function will be called
+        instead.
+
+        This function can be used when the target being swapped is used by
+        unrelated code. For example, os.path.exists() is used extensively by
+        psutil to manage processes. If we swap it to _always_ return True or
```

Comment:
If this is the main application, I wonder if you can narrow the API of this utility somehow so that it only handles situations like that.

Reply:
Obsolete.